src/repos/ArticleRepo.ts

```diff
@@ -1,28 +1,51 @@
 import Filter from 'bad-words';
 import { ObjectId } from 'mongodb';
 import { Article, ArticleModel } from '../entities/Article';
-import { DEFAULT_LIMIT, MAX_NUM_DAYS_OF_TRENDING_ARTICLES } from '../common/constants';
+import { DEFAULT_LIMIT, MAX_NUM_DAYS_OF_TRENDING_ARTICLES, IS_FILTER_ACTIVE, FILTERED_WORDS} from '../common/constants';
 import { PublicationModel } from '../entities/Publication';
 
+function isArticleFiltered(article: Article){
+  if(IS_FILTER_ACTIVE){
+    if (article.isFiltered){ //If the body has been checked already in microservice
+      return article.isFiltered; 
+    }
+    const filter = new Filter({ list: FILTERED_WORDS });
+    return filter.isProfane(article.title);
+  }
+  return false;
+}
+
 const getArticleByID = async (id: string): Promise<Article> => {
-  return ArticleModel.findById(new ObjectId(id));
+  return ArticleModel.findById(new ObjectId(id)).then(
+    (article) => {
+      if (!isArticleFiltered(article)){
+        return article;
+      }
+    }
+  );
 };
 
 const getArticlesByIDs = async (ids: string[]): Promise<Article[]> => {
   return Promise.all(ids.map((id) => ArticleModel.findById(new ObjectId(id)))).then((articles) => {
     // Filter out all null values that were returned by ObjectIds not associated
     // with articles in database
-    return articles.filter((article) => article !== null);
+    return articles.filter((article) => article !== null && !isArticleFiltered(article));
   });
 };
 
 const getAllArticles = async (limit = DEFAULT_LIMIT): Promise<Article[]> => {
-  return ArticleModel.find({}).limit(limit);
+  return ArticleModel.find({}).limit(limit).then((articles) =>
```

Comment:
what is this limit for? are we doing pagination eventually? Volume's load times do be slow

Reply:
sad, we hate slow load times 
TT

Reply:
it's probably just cuz there's a lot of filtering that it's slow

also, are there multiple containers being spun up in the docker compose file? having only 1 container could be another why it's slow

Reply:
@chalo2000 @alanna-zhou 
Yes, I did ask @kidzegeye to eventually implement pagination, but I'm not sure any of that code should be in this PR... If this change isn't relevant to pagination then ignore this comment.

Reply:
The limit code is from before the PR